lasagne/layers/transform.py

```diff
@@ -0,0 +1,200 @@
+import theano.tensor as T
+
+from .base import MergeLayer
+
+__all__ = [
+    "TransformerLayer",
+]
+
+
+class TransformerLayer(MergeLayer):
+    """
+    Spatial transformer layer [1]_.
+
+    Parameters
+    ----------
+    input : a :class:`Layer` instance
+        The input where the affine transformation is applied. This should
+        have convolution format, i.e (num_batch, channels, height, width).
+    localization_network: : a :class:`Layer` instance
+        The network that calculates the parameters of the affine
+        transformation. See the example for how to initialize to the identity
+        transform. Note that all parameters are clipped to lie in the range
+        ``[-1, 1]``, similar to [1]_.
```

Comment:
Does that mean you're clipping the values in the code? Wouldn't it be more efficient to just require the `localization_network` to output values in [-1, 1] in the first place?